src/lib.rs

```diff
@@ -79,6 +79,18 @@ where
 
         self.frontmatter
     }
+    /// Iterate over the FrontmatterExtractor iterator and stop after it has finished
+    /// searching for frontmatter. Returns the Iterator so you can do additional parsing
+    /// on the rest of the document. The optional first H1 title header and frontmatter
+    /// code block have been consumed
+    pub fn iterate_and_return(mut self) -> FrontmatterExtractor<'a, T> {
```

Comment:
Since this function always returns self, why wouldn't this take `&mut self`?

Reply:
TBH I just copied the extract function, &mut self works just fine here. I'll update